### PR TITLE
Makes Clonepods and Slime Processors Suck

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Makes clonepods and and slime processors suck...well, more.

Slime processor comes from TG: https://github.com/tgstation/tgstation/pull/16773 @Iamgoofball 

- Clonepods will automatically suck in nearby meat, giving the cloner biomass+deleting the meat
- Slime Processors will automatically, one at a time, suck in dead slimes (you still have to manually start the processor, of course)

:cl: Fox McCloud
Add: Clonepods will automatically suck in nearby meat
Add: Slime processors will automatically suck in nearby dead slimes
/:cl:

### CODE REVIEW

**Comment by Bigtimetripod** [comment]
Aww... i cannot punish golems no more to load in slimes one by one...

**Comment by IcyV** [comment]
Now I need to press even fewer buttons than already required to do Xenobio!

**Comment by Davdichan** [comment]
Is meat one by one or all at once? Also does range 1 include diagonal (I.E. if I put two cloning pods next to each other, will it evenly eat the meat or only the one directly adjacent and not diagonal from the meat?)

**Comment by LittleBigKid2000** [comment]
No ERP, Fox

**Comment by Krausus** [comment]
Sucking in nearby meat sounds like it would be incredibly dangerous. Like, "if this gets emagged, it'll occasionally yank limbs off nearby people" dangerous. Feels like there's an opportunity being missed here...

**Comment by Fox-McCloud** [comment]
The meat is all at once, in a 3x3 grid around the clone pod.

>Aww... i cannot punish golems no more to load in slimes one by one

Still could; it only process 1 dead slime every 2 seconds, and if you wanted to "punish" your golems, you could put it two tiles away from the processor =p

**Comment by Davdichan** [comment]
So multiple pods won't share meat evenly. Gotcha.

**Comment by taukausanake** [comment]
So being more lazy then? No thanks, I like actually doing things

**Comment by Da-Dman234** [comment]
I can maybe understand it's satisfying to process slimes by yourself, but sticking 20+ slabs of meat in the cloner? I fail to see how that's the least bit enjoyable.

**Comment by Spacemanspark** [comment]
Excuse me, @Da-Dman234, but you will respect my fetish of putting meat into a cloning pod sexually.

**Comment by KasparoVy** [comment]
"Add: Slime processors will automatically suck in nearby dead **slimse**"

**Comment by taukausanake** [comment]
@Spacemanspark No ERP. And if you must then you should work for it before you get BSA'd

**Comment by LittleBigKid2000** [comment]
An emagged slime processor should be able to suck in dead slime people

**Comment by Bigtimetripod** [comment]
@LittleBigKid2000

 Think of the slime people meta then! Every round start all slime people gather infront of xenobio and start riots to space slime processor!

**Comment by ProperPants** [comment]
Why was it "unironically buffing science" for @monster860 to make stock parts build faster, but this isn't buffing science? Both were really good, though, but isn't this technically a buff?

**Comment by taukausanake** [comment]
Because no one cares for Xenobiology. You should rather be asking if this buffs cloning at all which sounds retarded now that I've typed it...

**Comment by IcyV** [comment]
Xenobio is like a 5 step process as is with the computers added in. (Wasn't that controversial anyway?) This just makes it that much less interesting to do. I've already seen the amount of people that do Xenobio die down a ton since those were added.

**Comment by monster860** [comment]
@Fox-McCloud We will never forget that comment, and we will never let you forget it.

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -292,6 +292,15 @@
 
 //Grow clones to maturity then kick them out.  FREELOADERS
 /obj/machinery/clonepod/process()
+	var/show_message = 0
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/meat in range(1, src))
+		if(meat)
```
Comment (PJB3005) [inline code/game/machinery/cloning.dm]: That's pretty redundant.

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -292,6 +292,15 @@
 
 //Grow clones to maturity then kick them out.  FREELOADERS
 /obj/machinery/clonepod/process()
+	var/show_message = 0
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/meat in range(1, src))
```
Comment (PJB3005) [inline code/game/machinery/cloning.dm]: Maybe check adjacency?

File: code/game/machinery/cloning.dm [hunk 3]
```diff
@@ -292,6 +292,15 @@
...
 /obj/machinery/clonepod/process()
+	var/show_message = 0
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/meat in range(1, src))
+		if(meat)
+			qdel(meat)
+			biomass += 50
+			show_message = 1
+	if(show_message)
+		visible_message("[src] sucks in and processes the nearby biomass.")
+		show_message = 0
```
Comment (PJB3005) [inline code/game/machinery/cloning.dm]: Seems redundant since this var isn't referenced beyond this point.

File: code/game/machinery/cloning.dm [hunk 4]
```diff
@@ -292,6 +292,15 @@
 
 //Grow clones to maturity then kick them out.  FREELOADERS
 /obj/machinery/clonepod/process()
+	var/show_message = 0
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/meat in range(1, src))
+		if(meat)
+			qdel(meat)
+			biomass += 50
```
Comment (PJB3005) [inline code/game/machinery/cloning.dm]: This is a magic number.

File: code/game/machinery/kitchen/processor.dm [hunk 5]
```diff
@@ -29,6 +29,29 @@
...
 		rating_speed = M.rating
 
+/obj/machinery/processor/process()
+	..()
+	// The irony
+	// To be clear, if it's grinding, then it can't suck them up
+	if(processing)
+		return
+	var/mob/living/carbon/slime/picked_slime
+	for(var/mob/living/carbon/slime/slime in range(1,src))
```
Comment (PJB3005) [inline code/game/machinery/kitchen/processor.dm]: Use `orange()`, eliminate the check below.

File: code/game/machinery/kitchen/processor.dm [hunk 6]
```diff
@@ -29,6 +29,29 @@
...
+		if(slime.loc == src)
+			continue
+		if(istype(slime, /mob/living/carbon/slime))
```
Comment (PJB3005) [inline code/game/machinery/kitchen/processor.dm]: Redundant since the loop won't get non-slimes.

File: code/game/machinery/kitchen/processor.dm [hunk 7]
```diff
@@ -29,6 +29,29 @@
...
+			if(slime.stat)
+				picked_slime = slime
+				break
+	if(!picked_slime)
+		return
+	var/datum/food_processor_process/P = select_recipe(picked_slime)
+	if (!P)
+		return
+
+	visible_message("[picked_slime] is sucked into [src].")
```
Comment (PJB3005) [inline code/game/machinery/kitchen/processor.dm]: Should be `\the [src]`

File: code/game/machinery/kitchen/processor.dm (hunk 5 above)
Comment (PJB3005) [inline code/game/machinery/kitchen/processor.dm]: Oh and check adjacency.

Comment (tigercat2000) [inline code/game/machinery/kitchen/processor.dm]: why check adjacency @PJB3005

Comment (Fox-McCloud) [inline code/game/machinery/kitchen/processor.dm]: That's not how its meant to work, either. It's meant to work for the turf the processor is on, it's just meant to exclude slimes *inside* the processor.

**Pull request merged**